src/foam/nanos/auth/login/ClientLoginService.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth.login',
+  name: 'ClientLoginService',
+
+  imports: [
+    'auth',
+    'ctrl',
+    'defaultUserLanguage',
+    'emailVerificationService',
+    'loginSuccess',
+    'notify',
+    'stack',
+    'subject',
+    'wizardController',
+    'wizardletId',
+    'wizardlets'
+  ],
+
+  requires: [
+    'foam.log.LogLevel',
+    'foam.nanos.auth.DuplicateEmailException',
+    'foam.nanos.auth.login.SignIn',
+    'foam.nanos.auth.UnverifiedEmailException',
+    'foam.nanos.auth.User',
+    'foam.u2.stack.StackBlock'
+  ],
+
+  messages: [
+    { name: 'SIGNIN_ERR', message: 'There was an issue logging in' },
+    { name: 'SIGNUP_ERR', message: 'There was a problem creating your account' },
+    { name: 'SIGNUP_SUCCESS_MSG', message: 'Account successfully created' },
+    { name: 'SIGNUP_SUCCESS_TITLE', message: 'Success' },
+  ],
+
+  methods: [
+    {
+      name: 'signin',
+      code: async function(x, data, wizardFlow) {
+        var analyticable = foam.nanos.analytics.Analyticable.isInstance(data);
+        try {
+          ctrl.groupLoadingHandled = true;
```

Comment:
This is referencing the global ctrl and not the import, should be `this.ctrl.groupLoadingHandled`

Reply:
corrected.